### ISSUE TITLE
`indent`: Unnecessary indentation is added when the type argument contains a function

### ISSUE THREAD

**Issue by castella-cake** [eslint-stylistic/eslint-stylistic#909]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Unnecessary indentation is added when a generic function call contains a function as a type argument.

Expected (Original code):
```ts
genericFunction<
  () => void
>(
  () => {
    console.log("Test");
  }
);
```

After auto-fixed:
```ts
genericFunction<
  () => void
>(
    () => {
      console.log("Test");
    }
    );
```

Notably, this only occurs when the type argument is written on multiple lines.
For example, the following code does not cause a lint error:
```ts
genericFunction<() => void>(
  () => {
    console.log("Test");
  }
)
```

This might be an edge case, since it can be easily avoided by predefining the type.
Thanks for maintaining the project!

### Reproduction

https://stackblitz.com/edit/github-w4u4dpcu

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
